client/src/Contexts/GameContext.jsx

```diff
@@ -9,11 +13,34 @@ import { ShifumiWeaponObject, ShifumiResultObject } from "../Enums/Shifumi";
 export const GameContext = createContext({});
 
 export function GameContextProvider({ children }) {
-  const [userSelection, setUserSelection] = useState();
+  const webSocket = useWebSocketContext();
+  const { user } = useUserContext();
+
+  const [userSelection, setUserSelection] = useState('');
+  const [all, setAll] = useState([])
+
+  const [opponent, setOpponent] = useState('')
+
   const [cpuSelection, setCpuSelection] = useState();
   const [userMatchResult, setUserMatchResult] = useState();
   const [score, setScore] = useState(0);
 
+  console.log(all)
+  console.log(userSelection)
+
+
+  useEffect(() => {
+    webSocket.on("selection", ({ name: user, weapon: userSelection }) => {
```

Comment:
```suggestion
    webSocket.on(WebsocketEvent.SELECTION, ({ name: user, weapon: userSelection }) => {
```
ça pourrait être pratique d'avoir une enum ici

Reply:
Je veux bien que tu m'expliques pourquoi alors !